From: YOKOTA Hiroshi <yokota.hgml@gmail.com>
Date: Fri, 17 Nov 2023 15:34:02 +0900
Subject: Revert "Restore garbage removal on podofo save as we now bundle a
 version of podofo with the fix"

Forwarded: not-needed

This reverts commit dec4f9f82e6edc3ed7b5ea8dcce6b2ef2ea59cea.
---
 src/calibre/utils/podofo/global.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/calibre/utils/podofo/global.h b/src/calibre/utils/podofo/global.h
index d819afb..74bc124 100644
--- a/src/calibre/utils/podofo/global.h
+++ b/src/calibre/utils/podofo/global.h
@@ -130,7 +130,8 @@ object_as_reference(const PdfObject *o) {
 }
 
 // NoMetadataUpdate needed to avoid PoDoFo clobbering the /Info and XMP metadata with its own nonsense
-static const PdfSaveOptions save_options = PdfSaveOptions::NoMetadataUpdate;
+// NoCollectGarbage needed to avoid slowdowns on writing, see https://github.com/podofo/podofo/issues/108
+static const PdfSaveOptions save_options = PdfSaveOptions::NoMetadataUpdate | PdfSaveOptions::NoCollectGarbage;
 
 class PdfReferenceHasher {
     public:
